CHANGELOG.md

```diff
@@ -1,10 +1,840 @@
+Last releases: [[2.4.0] - 2025-04-14](#240---2025-04-14), [[2.3.0] - 2024-11-28](#230---2024-11-28), [[2.2.0] - 2024-01-17](#220---2024-01-17)
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
+## [2.4.0] - 2025-04-14
+
+This release is compatible with Coq versions 8.19, 8.20 and Rocq version 9.0.
+
+The contributors to this version are:
+
+Alessandro Bruni, Cyril Cohen, Enrico Tassi, Erik Martin-Dorel, Kazuhiko Sakaguchi, Kimaya Bedarkar, Laurent Théry, Pierre Pomeret-Coquot, Pierre Roux, Quentin Vermande, Reynald Affeldt, Mitsuharu Yamamoto, Yves Bertot
+
+### Added
+
+- in `ssralg.v`
+  + Semimodule and semialgebra structures `lSemiModType`, `lSemiAlgType`, `semiAlgType`, `comSemiAlgType`, `subLSemiModType`, `subLSemiAlgType`, and `subSemiAlgType`.
+  + Scaling-morphism structures `GRing.Scale.preLaw` and `GRing.Scale.semiLaw`.
+  + Mixins `Nmodule_isLSemiModule`, `LSemiModule_isLSemiAlgebra`, `GRing.Scale.isPreLaw`, `GRing.Scale.isSemiLaw`, `LSemiAlgebra_isSemiAlgebra`, `isSubLSemiModule`
+  + Factories `LSemiModule_isLmodule`, `isSemilinear`, `LSemiAlgebra_isComSemiAlgebra`, `SubNmodule_isSubLSemiModule`, `SubNzSemiRing_SubLSemiModule_isSubLSemiAlgebra`, `SubLSemiAlgebra_isSubSemiAlgebra`, `SubChoice_isSubLSemiModule`
+  + Definitions `semilinear_for`, `subsemimod_closed`
+  + Notations `semilinear`, `semiscalar`, `[SubNmodule_isSubLSemiModule of V by <:]`, `[SubChoice_isSubLSemiModule of V by <:]`, `[SubSemiRing_SubLSemiModule_isSubLSemiAlgebra of V by <:]`, `[SubLSemiAlgebra_isSubSemiAlgebra of V by <:]`
+  + Lemmas `subsemimod_closedD`, `subsemimod_closedZ`, `submod_closed_semi`, `additive_semilinear`, `scalable_semilinear`, `semilinear_linear`, `semilinearP`, `semilinearPZ`, `can2_semilinear`, `semiscalarP`, `subsemimodClosedP`
+    ([#1125](https://github.com/math-comp/math-comp/pull/1125),
+    by Kazuhiko Sakaguchi).
+  + mixin `Nmodule_isPzSemiRing`
+  + definition `pzSemiRingType`
+  + factory `isPzSemiRing`
+  + mixin `PzSemiRing_isNonZero`
+  + definition `pzRingType`
+  + factories `Zmodule_isPzRing`, `isPzRing`
+  + mixin `PzSemiRing_hasCommutativeMul`
+  + definition `comPzSemiRingType`
+  + factory `Nmodule_isComPzSemiRing`
+  + definition `comPzRingType`
+  + factories `PzRing_hasCommutativeMul`, `Zmodule_isComPzRing`
+  + mixin `isSubPzSemiRing`
+  + definition `subPzSemiRingType`
+  + factory `SubNmodule_isSubPzSemiRing`
+  + definition `subComPzSemiRingType`
+  + factory `SubPzSemiRing_isSubComPzSemiRing`
+  + definition `subPzRingType`
+  + factory `subZmodule_isSubPzRing`
+  + definition `subComPzRingType`
+  + factories `SubPzRing_isSubComPzRing`, `SubChoice_isSubPzSemiRing`,
+    `SubChoice_isSubComPzSemiRing`, `SubChoice_isSubPzRing`,
+    `SubChoice_isSubComPzRing`
+    ([#1319](https://github.com/math-comp/math-comp/pull/1319),
+    by Tragicus).
```

Comment:
```suggestion
    by Quentin Vermande).
```
Or it can be the other way around, but it should probably be uniformized anyway. @Tragicus

Reply:
(all the other entries don't mention any PR)

Reply:
We said we could remove the name, but the link definitely remains worthwile data.